lib/chai/core/assertions.js

```diff
@@ -3013,19 +3025,18 @@ function closeTo(expected, delta, msg) {
     , flagMsg = flag(this, 'message')
     , ssfi = flag(this, 'ssfi');
 
-  new Assertion(obj, flagMsg, ssfi, true).is.a('number');
-  if (typeof expected !== 'number' || typeof delta !== 'number') {
-    flagMsg = flagMsg ? flagMsg + ': ' : '';
-    var deltaMessage = delta === undefined ? ", and a delta is required" : "";
-    throw new AssertionError(
-        flagMsg + 'the arguments to closeTo or approximately must be numbers' + deltaMessage,
-        undefined,
-        ssfi
-    );
-  }
+  new Assertion(obj, flagMsg, ssfi, true).is.numeric;
+  let message =  'A `delta` value is required for `closeTo`';
+  if (!delta) throw new AssertionError(flagMsg ? `${flagMsg}: ${message}` : message, undefined, ssfi);
+  new Assertion(delta, flagMsg, ssfi, true).is.numeric;
+  message =  'A `expected` value is required for `closeTo`';
+  if (!delta) throw new AssertionError(flagMsg ? `${flagMsg}: ${message}` : message, undefined, ssfi);
```

Comment:
Can you have a delta of `0`? 👀 Do we need to compare explicitly against undefined rather than a falsy check?

Reply:
Could be but then we wouldn't be comparing to see if the two values are close since they'd need to be the same.

I can't think of a reason why someone would do that but adding a test and making this a specific `undefined` check is good in case someone does want to do that in future for whatever reason.